File: python/util/pandigital_test.py
import unittest, pandigital


class TestPandigital(unittest.TestCase):
  def test_pandigital(self):
    self.assertEqual(pandigital.is_pandigital(134706, exclude_zero=False),
                     True)
    self.assertEqual(pandigital.is_pandigital(134706, exclude_zero=True),
                     False)
    self.assertEqual(pandigital.is_pandigital('134706', exclude_zero=False),
                     True)
    self.assertEqual(pandigital.is_pandigital('134706', exclude_zero=True),
                     False)


if __name__ == '__main__':
  unittest.main()

File: javascript/util/pandigital.test.js
const isPandigital = require(`./pandigital`);

test(`expects that 134706 is pandigital`, () => {
  expect(isPandigital(134706)).toBe(true);
});

test(`expects that 134706 is not pandigital when you exclude zero`, () => {
  expect(isPandigital(134706, { excludeZero: true })).toBe(false);
});

test(`expects that '134706' is pandigital`, () => {
  expect(isPandigital(`134706`)).toBe(true);
});

test(`expects that '134706' is not pandigital when you exclude zero`, () => {
  expect(isPandigital(`134706`, { excludeZero: true })).toBe(false);
});
